quarkchain/cluster/master.py

```diff
@@ -1063,11 +1063,12 @@ def __log_summary(self):
 
     async def __init_cluster(self):
         await self.__connect_to_slaves()
-        self.__log_summary()
-        if not self.__has_all_shards():
-            Logger.error("Missing some shards. Check cluster config file!")
-            return
-        await self.__setup_slave_to_slave_connections()
+        if not self.grpc_slave_pool:
```

Comment:
1. should check on `self.slave_pool`
2. `self.__connect_to_slaves()` and `self.__init_shards()` should also be under the condition check?

Reply:
`self.slave_pool` is generated by `self.__connect_to_slaves()`

Reply:
then how about check `self.cluster_config.get_slave_info_list()`